pykafka/producer.py

```diff
@@ -231,14 +231,38 @@ def _setup_owned_brokers(self):
                     self, partition.leader)
         return queued_messages
 
-    def stop(self):
-        """Mark the producer as stopped"""
-        self._running = False
+    def _stop_owned_brokers(self):
         self._wait_all()
         if self._owned_brokers is not None:
             for owned_broker in self._owned_brokers.values():
                 owned_broker.stop()
 
+    def get_queue_readers(self):
+        if not self._owned_brokers:
+            return []
+        return [owned_broker._queue_reader_worker for owned_broker in self._owned_brokers.values() if owned_broker.running]
+
+    def stop(self, wait=True):
```

Comment:
I'm not sure this extra kwarg is necessary. The current implementation of `stop()` assumes that there is no reason we'd want to avoid waiting, and I think it's fine to keep it that way for now. Additionally, keeping the function signature the same will make this change easier to get into a release since it won't need to be rolled into a minor version update.